src/utils/string_parsing.ts

```diff
@@ -209,6 +242,17 @@ function intToHex(num : number, size : number) : string {
  * @returns {string}
  */
 function utf8ToStr(data : Uint8Array) : string {
+  if (typeof window.TextDecoder === "function") {
+    try {
+      // TextDecoder use UTF-8 by default
+      const decoder = new TextDecoder();
+      return decoder.decode(data);
+    } catch (e) {
+      log.warn("Utils: could not use TextDecoder to parse UTF-8, " +
```

Comment:
Isn't it a risk that these logs would be emitted too often?

Reply:
This is a very rare occurrence as TextDecoder still has to be defined.

But yes, this can happen at each text segment, which could be too verbose considering the level.
Maybe a warning is too high level for such a minor issue.
You think I should use a debug log instead?

Reply:
And why not using a warnOnce? The user may just need to be warned once about the impossibility of using the TextDecoder.

Reply:
`warnOnce` does not go through the logger.
It was especially defined for deprecation warnings, which we want to display even when no log is displayed.

We could add a new kind of logging function for that but I don't think that the game is worth the candle !

Reply:
Oh yeah, I had forgotten! Then ok, maybe it isn't worth breaking your neck over.
So let's keep it as it is? You're, it is a rare occurrence. I think it should still be a warning instead a debug log.